aws-sdk-resources/lib/aws-sdk-resources/services/s3/object_copier.rb

```diff
@@ -12,6 +12,7 @@ def initialize(object, options = {})
       end
 
       def copy_from(source, options = {})
+        options[:copy_source_client] = source.client if source.is_a?(S3::Object)
```

Comment:
This could also be expressed as:

```ruby
options[:copy_source_client] ||= source.client if source.respond_to?(:client)
```

The `||=` is less important, the `respond_to?` check makes this a little more future proof. Currently we have not added support for `Aws::S3::ObjectVersion`, but this is likely to happen. No need to change this unless you feel inclined. If you don't, we will make the change when adding object version support.